### PR TITLE
 Make "dnu restore" accept multiple restore root dirs 

### PR DESCRIPTION
**Pull request by ChengTian**
parent https://github.com/aspnet/dnx/issues/1732

`dnu restore <dir1> <dir2> <dir3> ...` and dnu will restore for each path specified as an argument.

@davidfowl

### CODE REVIEW

File: src/Microsoft.Framework.PackageManager/Program.cs [hunk 1]
```diff
@@ -61,7 +61,9 @@ public int Main(string[] args)
             {
                 c.Description = "Restore packages";
 
-                var argRoot = c.Argument("[root]", "Root of all projects to restore. It can be a directory, a project.json, or a global.json.");
+                var argRoot = c.Argument("[root]",
+                    "Root of projects to restore. A value can be a path to directory/project.json/global.json. Accepts multiple values.",
+                    multipleValues: true);
```
Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Program.cs]: @Eilon , I bet you have a better idea about help text of the argument here. This argument can have multiple values:
```
dnu restore <path1> <path2> <path3>
```
The three paths are all values of the argument `[root]`

**Comment by davidfowl** [comment]
As part of this change can you make it so that we don't recurse if the current directory has a `project.json`?

File: src/Microsoft.Framework.PackageManager/Program.cs (hunk 1 above)
Comment (Eilon) [inline src/Microsoft.Framework.PackageManager/Program.cs]: Can the name of the "argument" be `[projects]` instead?

`List of projects and project folders to restore. The values can be: a path to project.json or global.json file, or a folder containing a project.json or global.json.`

Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Program.cs]: Yeah, `roots` is not a clearly defined concept. It's better to known concept `project`. Will change to `projects`

Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Program.cs]: I am going to make the change mentioned by @davidfowl below. So if there is no `project.json` in given folder, we recursively traverse all subfolders and restore for all `project.json` we see. So we need to revise the help info a little bit:
```
List of projects and project folders to restore. The values can be: a path to project.json or global.json file, or a folder having project.json in subfolders
```

Comment (Eilon) [inline src/Microsoft.Framework.PackageManager/Program.cs]: Slight tweak (also fixed typo from my earlier string): ` List of projects and project folders to restore. Each value can be: a path to a project.json or global.json file, or a folder to recursively search for project.json files.`

**Comment by ChengTian** [comment]
@davidfowl , I added that part in the last commit.

**Comment by davidfowl** [comment]
@ChengTian why is the build failing?

**Comment by ChengTian** [comment]
@davidfowl , build passed

**Comment by ChengTian** [comment]
@davidfowl , our CI will be fixed today. Please sign off so that Tooling team can integrate this feature and verify it.

**Comment by ChengTian** [comment]
@davidebbo , rebased and `build verify` locally

**Comment by davidebbo** [comment]
@ChengTian wrong David I think ;)

**Comment by Eilon** [comment]
@davidfowl !

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**